File: responses/YJ7PGK-HKS/longest_common_prefix.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(x, b)
  return 0 if x > b.size
  return 0 if x < 0
  return 1 if x == 0
  
  return (0 .. (b.size - x)).to_a.map { |i| b[i, x].reduce(&:*) }.max
end

File: responses/9T3V68-NFA/longest_common_prefix.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(X, B):
    if X > len(B) or B == []:
        return 0
    
    prod_arr = []
    for i in range(X):
        prod_arr.append(max(B))
        B.remove(max(B))
    
    product = 1 
    for number in prod_arr:
        product *= number
        
    return product
        
        